CRM/Core/CodeGen/Specification.php

```diff
@@ -704,9 +704,9 @@ public function getForeignKey(&$foreignXML, &$fields, &$foreignKeys, &$currentTa
    */
   public function getDynamicForeignKey(&$foreignXML, &$dynamicForeignKeys) {
     $foreignKey = [
-      'idColumn' => trim($foreignXML->idColumn),
-      'typeColumn' => trim($foreignXML->typeColumn),
-      'key' => trim($this->value('key', $foreignXML)),
+      'idColumn' => trim($foreignXML->idColumn ?? ''),
```

Comment:
I'm not sure if this PR is another test just to trigger the test system but I remember simplexml objects don't behave like normal. This would never be null unless $foreignXML itself is null because idColumn itself is an object even if it doesn't exist in the original xml. I think in the end it's ok, but I don't know if it's catching the situation you're trying to catch.

Reply:
I think it was mostly key that was the problem but I might have just been a bit overzealos with the protection

Reply:
What do you want to do with this one? If it is just "key", then I'd lean towards something narrower so as not to hide real errors. Like just replacing that line with `'key' => trim($this->value('key', $foreignXML, '')),`